src/core/graph/state/graphStateStore.ts

```diff
@@ -0,0 +1,79 @@
+import { defineStore } from 'pinia'
+import { customRef } from 'vue'
+
+import type { NodeLocatorId } from '@/types/nodeIdentification'
+
+interface NodeState {
+  hasError: boolean
+}
+
+interface SetNodeErrorCommand {
+  type: 'SetNodeError'
+  version: 1
+  nodeId: NodeLocatorId
+  hasError: boolean
+}
+
+interface ClearAllErrorsCommand {
+  type: 'ClearAllErrors'
+  version: 1
+}
+
+type GraphStateCommand = SetNodeErrorCommand | ClearAllErrorsCommand
+
+export const useGraphStateStore = defineStore('graphState', () => {
+  const nodes = new Map<NodeLocatorId, NodeState>()
+
+  let revision = 0
+  const stateRef = customRef<number>((track, trigger) => ({
+    get() {
+      track()
+      return revision
+    },
+    set() {
+      revision++
+      trigger()
+    }
+  }))
```

Comment:
Question: Could you check to see if this works the same? 
The setter is currently ignoring the value passed in on line 57, right?
```suggestion
  const stateRef = ref(0)
```